File: java/src/main/java/org/sparky/model/builders/ValueBuilder.java
package org.sparky.model.builders;

import org.sparky.model.Bit;
import org.sparky.model.Rule;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by lloyd on 17/10/2015.
 */
public class ValueBuilder implements EvaluatableBuilder {

    private final List<Bit> bits = new ArrayList<>();

    @Override
    public ValueBuilder withBit(Bit bit) {
        bits.add(bit);
        return this;
    }

    public Rule build() {
        return new Rule(bits);
    }
}

File: net/Sparky/Model/Builder/ValueBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sparky.Model.Builder
{
    public class ValueBuilder : IEvaluatableBuilder
    {
        private readonly List<IPartial> _bits = new List<IPartial>();

        public Rule Build()
        {
            return new Rule(_bits);
        }

        public IEvaluatableBuilder WithPartial(IPartial partial)
        {
            _bits.Add(partial);
            return this;
        }
    }
}
